# HG changeset patch
# User jwe
# Date 784404255 0
#      Wed Nov 09 18:04:15 1994 +0000
# Node ID 4733b101ce3fbae559dcd444333818547cd7acec
# Parent  54921d0802953f53165f4f8a9c0e5ba3e9fbde44
[project @ 1994-11-09 18:04:15 by jwe]
Initial revision

diff --git a/scripts/miscellaneous/tic.m b/scripts/miscellaneous/tic.m
new file mode 100644
--- /dev/null
+++ b/scripts/miscellaneous/tic.m
@@ -0,0 +1,35 @@
+# Copyright (C) 1994 John W. Eaton
+#
+# This file is part of Octave.
+#
+# Octave is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2, or (at your option) any
+# later version.
+#
+# Octave is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Octave; see the file COPYING.  If not, write to the Free
+# Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+
+function tic ()
+
+# usage: tic
+#
+# Set a wall-clock timer.
+#
+# See also: toc, clock, etime, cputime
+
+  if (nargin != 0)
+    warning ("tic: ignoring extra arguments");
+  endif
+
+  global _time_tic_called;
+
+  _time_tic_called = clock ();
+
+endfunction
diff --git a/scripts/miscellaneous/toc.m b/scripts/miscellaneous/toc.m
new file mode 100644
--- /dev/null
+++ b/scripts/miscellaneous/toc.m
@@ -0,0 +1,41 @@
+# Copyright (C) 1994 John W. Eaton
+#
+# This file is part of Octave.
+#
+# Octave is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2, or (at your option) any
+# later version.
+#
+# Octave is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Octave; see the file COPYING.  If not, write to the Free
+# Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+
+function secs = toc ()
+
+# usage: toc
+#
+# Return the difference between the current wall-clock time and the
+# time that the function tic () was last called, in seconds.
+#
+# See also: tic, clock, etime, cputime
+
+  if (nargin != 0)
+    warning ("toc: ignoring extra arguments");
+  endif
+
+  global _time_tic_called;
+
+  if (exist ("_time_tic_called"))
+    secs = etime (clock (), _time_tic_called);
+  else
+    warning ("toc called before timer set");
+    secs = [];
+  endif
+
+endfunction
